pkgs/development/libraries/libidn2/hack.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, lib, libidn2, libunistring, runCommandLocal, patchelf }:
+# Construct a copy of libidn2.* where all (transitive) libc references (in .bin)
+# get replaced by a new one, so that there's no reference to bootstrap tools.
+runCommandLocal
+  "${libidn2.pname}-${libidn2.version}"
+  {
+    outputs = [ "bin" "dev" "out" ];
+    buildInputs = [ patchelf ];
+
+    passthru = {
+      inherit (libidn2) out info devdoc; # no need to touch these store paths
```

Comment:
This leads to unexpected results when requesting libidn2 from the CLI, which generally works with `drvPath` and its libstore-level outputs, rather than the outputs attribute. I'm proposing to redefine "package" in Nix here https://github.com/NixOS/nix/issues/6507.

Reply:
Presumably it works fine when used in expressions though.

It is only a bit of a problem when doing `nix-build -A libidn2`.

Reply:
What are you getting from that command?  I'm getting the right -bin output, i.e. the one without boostrap tools in closure.  (also with `nix build`)

Reply:
`libidn2.all` is a bit confusing, though I don't think it has so much usage.  I also tried to make that not too bad... so you're missing `info` and `devdoc`, and `out` only differs by adding a symlink layer.

Reply:
Right, it's more of a problem with the new CLI's output selector syntax, which operates at the store level (via `drvPath`) rather than the expression level.

Reply:
```
$ nix-build -A libidn2.info
/nix/store/klbv14mif5q34nqj04v2bgv9d7qv2pk6-libidn2-2.3.2-info

$ nix build .#libidn2^info --print-out-paths 
warning: Git tree '/home/user/h/nixpkgs' is dirty
error: derivation '/nix/store/zy3r4mqqb8wsljg1cfdnw02mqplaqw8n-libidn2-2.3.2.drv' does not have wanted outputs 'info'
```

Reply:
Oh I forgot to mention that stuff like `libidn2.info.bin` is also broken by such overrides, and I don't know a nice way of fixing that.  (But again, I think it's rarely used.)